docs/events/glides/com.mbta.ctd.glides.vehicle_trip_assignments.v1.mdx

```diff
@@ -0,0 +1,167 @@
+- `revenue` (`"revenue"` | `"nonrevenue"`): Whether the trip is a revenue trip.
+- `scheduled` (`"scheduled"` | `"added"`): Whether this is a trip that Glides has in its schedule or that Glides added.
+
+This is a different structure than the [`TripKey` used in `trips_updated`](com.mbta.ctd.glides.trips_updated.v1.mdx#tripkey). Added and scheduled trips use the same format, and are differentiated by the `scheduled` field instead of by using different formats.
+
+- If `scheduled` is `"scheduled"`, then this TripKey's `tripId` will equal the `trips_updated` TripKey's `tripId`.
+- If `scheduled` is `"added"`, then this TripKey's `tripId` will equal the `trips_updated` TripKey's `glidesId`.
+- (In both cases, the `serviceDate` must also match in order to know the two TripKeys refer to the same trip.)
+
+The difference with the TripKey in `trips_updated` is because `trips_updated` has to give enough information about a trip that a consumer can understand it even without recognizing the trip ID, while `vehicle_trip_assignments` assumes that you can look up information in `trips_updated` or in `GTFS`, so doesn't give as much information as possible about the trip. (Though consumers should keep in mind that order of events between two event streams is not guaranteed, and a trip that's referenced in this feed may not have appeared in `trips_updated`.)
+
+For scheduled trips, the `tripId` is intended to line up to the trip IDs in GTFS most of the time, because they're derived from the same source (HASTUS's `trp_int_number`). However, Glides may be using a slightly different schedule than any consumer (due to disruptions, for example), so matching GTFS is not guaranteed. If a consumer sees a `tripId` in this feed that also appears in GTFS on a trip on the same service date, then it refers to the same trip.
+
+For added trips, Glides makes up a new ID, which is kept consistent for any subsequent updates to the same added trip. Glides tends to prefix trip IDs for added trips with the string `"ADDED-"` but consumers MUST NOT assume any particular format, or assume that they can tell an ID for an added trip apart from a scheduled trip ID by the string alone. That's what the `scheduled` field is for.
+
+The added/scheduled distinction is according to the schedule in Glides. If Glides has a different schedule than RTR or GTFS, then a trip that Glides thinks is scheduled may not match up with scheduled trip in GTFS, and RTR (or other consumers) may decide that a trip that was added in Glides lines up close enough to correspond to a scheduled trip in GTFS.
+
+Revenue is included because this is the most important information that Glides can provide to riders about a train, and its presence will make it less likely for a consumer of this feed to incorrectly interpret a nonrevenue train as being in service. Especially since nonrevenue trips are not in GTFS and their IDs are likely to not be recognized by a consumer.
```

Comment:
> What if the trip isn't in GTFS and hasn't shown up in `trips_updated`, what would RTR do?

Why would it not have shown up in `trips_updated` if it had already been marked as non-revenue? I can understand maybe a veeeery slight delay between one message and the other if some trainsheet edits happen roughly simultaneously (say someone changed both the revenue status and the car numbers in one edit), but I don't think that would be enough lag to create much of a problem.

Reply:
I'll echo [my comment on the thread about which fields should be nullable](https://github.com/mbta/schemas/pull/9#discussion_r1847104346): these events are for providing information about trip - vehicle assignments, not for providing other information about trips. I think we're really over-complicating this by trying to make the `vehicle_trip_assignment` events do things that other events defined in the schema already do.

Reply:
As discussed on the call, revenue was removed (22245e1). If RTR has trouble finding info in trips_updated (because of the delay, or because the trip was never edited), it should log that, and we could consider adding revenue back (as a data field, not as part of the `TripKey`).

Glides should try to ensure that any trip updates get sent before this event, so the trip can be referenced when RTR consumes this event.